Tests.App/Services/MutagenControllerTest.cs

```diff
@@ -0,0 +1,117 @@
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using Coder.Desktop.App.Services;
+
+namespace Coder.Desktop.Tests.App.Services;
+
+[TestFixture]
+public class MutagenControllerTest
+{
+    [OneTimeSetUp]
+    public async Task DownloadMutagen()
+    {
+        var ct = new CancellationTokenSource(TimeSpan.FromSeconds(60)).Token;
+        var scriptDirectory = Path.GetFullPath(Path.Combine(TestContext.CurrentContext.TestDirectory,
+            "..", "..", "..", "..", "scripts"));
+        var process = new Process();
+        process.StartInfo.FileName = "powershell.exe";
+        process.StartInfo.UseShellExecute = false;
+        process.StartInfo.Arguments = $"-ExecutionPolicy Bypass -File Get-Mutagen.ps1 -arch {_arch}";
+        process.StartInfo.RedirectStandardError = true;
+        process.StartInfo.RedirectStandardOutput = true;
+        process.StartInfo.WorkingDirectory = scriptDirectory;
+        process.Start();
+        var output = await process.StandardOutput.ReadToEndAsync(ct);
+        TestContext.Out.Write(output);
+        var error = await process.StandardError.ReadToEndAsync(ct);
+        TestContext.Error.Write(error);
+        Assert.That(process.ExitCode, Is.EqualTo(0));
+        _mutagenBinaryPath = Path.Combine(scriptDirectory, "files", $"mutagen-windows-{_arch}.exe");
+    }
+
+    [SetUp]
+    public void CreateTempDir()
+    {
+        _tempDirectory = Directory.CreateTempSubdirectory(GetType().Name);
+        TestContext.Out.WriteLine($"temp directory: {_tempDirectory}");
+    }
+
+    private readonly string _arch = RuntimeInformation.ProcessArchitecture switch
+    {
+        Architecture.X64 => "x64",
+        Architecture.Arm64 => "arm64",
+        // We only support amd64 and arm64 on Windows currently.
+        _ => throw new PlatformNotSupportedException(
+            $"Unsupported architecture '{RuntimeInformation.ProcessArchitecture}'. Coder only supports x64 and arm64."),
+    };
+
+    private string _mutagenBinaryPath;
+    private DirectoryInfo _tempDirectory;
+
+    [Test(Description = "Shut down daemon when no sessions")]
+    [CancelAfter(30_000)]
+    public async Task ShutdownNoSessions(CancellationToken ct)
+    {
+        // NUnit runs each test in a temporary directory
+        var dataDirectory = _tempDirectory.FullName;
+        await using var controller = new MutagenController(_mutagenBinaryPath, dataDirectory);
+        await controller.Initialize(ct);
+        var logPath = Path.Combine(dataDirectory, "daemon.log");
```

Comment:
This doesn't seem to check that the daemon isn't running after initialize like the test name suggests

Reply:
Seems like you'd check for a single log line matching the line below and no `daemon.lock` or something

Reply:
I've added a check that we can get a lock on the `daemon.lock`.